sklearn/gaussian_process/gpr.py

```diff
@@ -241,8 +241,13 @@ def obj_func(theta, eval_gradient=True):
         # of actual query points
         K = self.kernel_(self.X_train_)
         K[np.diag_indices_from(K)] += self.alpha
-        self.L_ = cholesky(K, lower=True)  # Line 2
-        self.alpha_ = cho_solve((self.L_, True), self.y_train_)  # Line 3
+        try:
+            self.L_ = cholesky(K, lower=True)  # Line 2
+            self.alpha_ = cho_solve((self.L_, True), self.y_train_)  # Line 3
+        except np.linalg.LinAlgError:
```

Comment:
Are there other potential causes for `LinAlgError` here? Why does the try block need to include two (not one) statements?